app/scripts/migrations/096.ts

```diff
@@ -0,0 +1,66 @@
+import { cloneDeep } from 'lodash';
+
+type MetaMaskState = Record<string, unknown>;
+type VersionedState = {
+  meta: { version: number };
+  data: MetaMaskState;
+};
+
+export const version = 96;
+
+/**
+ * This migration removes obsolete NetworkController state properties.
+ *
+ * @param originalVersionedState - Versioned MetaMask extension state, exactly what we persist to dist.
+ * @param originalVersionedState.meta - State metadata.
+ * @param originalVersionedState.meta.version - The current state version.
+ * @param originalVersionedState.data - The persisted MetaMask state, keyed by controller.
+ * @returns Updated versioned of MetaMask extension state.
+ */
+export async function migrate(
+  originalVersionedState: VersionedState,
+): Promise<VersionedState> {
+  const updatedVersionedState = cloneDeep(originalVersionedState);
+
+  updatedVersionedState.meta.version = version;
+  updatedVersionedState.data = transformState(updatedVersionedState.data);
+
+  return updatedVersionedState;
+}
+
+function transformState(originalState: MetaMaskState): MetaMaskState {
+  const updatedState =
+    filterOutObsoleteNetworkControllerStateProperties(originalState);
+
+  return updatedState;
+}
+
+function filterOutObsoleteNetworkControllerStateProperties(
+  state: MetaMaskState,
+): MetaMaskState {
+  // See https://github.com/MetaMask/core/blob/35d887664189110554df1516950424de4171b036/packages/network-controller/src/NetworkController.ts#L360
```

Comment:
Great idea to include this comment! I think we'll want to use this link instead though:

```suggestion
  // See https://github.com/MetaMask/core/blob/%40metamask/network-controller%4010.3.1/packages/network-controller/src/NetworkController.ts#L336
```

That's the type for `@metamask/network-controller@10.3.1`, which is what v10.35.0 is using.

Reply:
Some of the properties differ from that commit as well - we'll want to ensure the migration is written for this version of the network controller. These newer properties won't be in v10.35.1